src/GAP/iso_oscar_gap.jl

```diff
@@ -30,10 +30,42 @@ end
 #
 ################################################################################
 
+@attributes Nemo.NmodRing # TODO: port this to Nemo
+@attributes Nemo.FmpzModRing # TODO: port this to Nemo
+
+# Assume that `RO` and `RG` are residue rings of the same size
+# in Oscar and GAP, respectively.
+function _iso_oscar_gap_residue_ring_functions(RO::Union{Nemo.NmodRing, Nemo.FmpzModRing}, RG::GAP.GapObj)
+   e = GAPWrap.One(RG)
+   f(x) = GAP.Obj(lift(x))*e
+
+   finv = function(x::GAP.Obj)
+     if GAP.Globals.IsFFE(x)
+       y = GAP.Globals.Int(x)
+     elseif GAP.Globals.IsZmodnZObj(x)
+       y = GAP.Globals.ExtRepOfObj(x)
+     else
+       error("GAP object $x is not supported")
+     end
```

Comment:
I think `GAP.Globals.Int` works in both cases.

Also, I notice that we don't perform any error checking here: so if the characteristic of `x` does not match that of `RO`, the conversion will silently work instead of producing an error. I think we already discussed this in the context of finite fields -- perhaps having a `check::Bool` argument somewhere would be useful?

Doesn't have to be in this PR, though, just wanted to point it out.

Reply:
Concerning error checking: The idea in #1092 was that for `MapFromFunc` objects, the checks are performed inside the `image` and `preimage` methods, not in the individual functions.